// todo-package/index.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

var TodoTask = /** @class */ (function () {
    /**
     * @param {?=} init
     */
    function TodoTask(init) {
        Object.assign(this, init);
    }
    return TodoTask;
}());
var TODOTASKS = [
    { taskID: 1, newTask: 'Task 1', geolocation: '', email: '', Image: '' },
    { taskID: 2, newTask: 'Task 2', geolocation: '', email: '', Image: '' },
    { taskID: 3, newTask: 'Task 3', geolocation: '', email: '', Image: '' },
    { taskID: 4, newTask: 'Task 4', geolocation: '', email: '', Image: '' },
    { taskID: 5, newTask: 'Task 5', geolocation: '', email: '', Image: '' }
];

var TodoTaskModule = /** @class */ (function () {
    function TodoTaskModule() {
    }
    /**
     * @return {?}
     */
    TodoTaskModule.forRoot = function () {
        return {
            ngModule: TodoTaskModule,
            providers: []
        };
    };
    TodoTaskModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule
                    ],
                    declarations: [
                        TodoTask
                    ],
                    exports: [
                        TodoTask
                    ]
                },] },
    ];
    /**
     * @nocollapse
     */
    TodoTaskModule.ctorParameters = function () { return []; };
    return TodoTaskModule;
}());

export { TodoTaskModule, TodoTask, TODOTASKS };
