// todo-package/index.js
import { Injectable, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HttpClient } from '@angular/common/http';
import 'rxjs/add/operator/map';

var TodoTask = /** @class */ (function () {
    /**
     * @param {?=} init
     */
    function TodoTask(init) {
        Object.assign(this, init);
    }
    return TodoTask;
}());
var TODOTASKS = [
    { taskID: 1, newTask: 'Task 1', geolocation: '', email: '', Image: '' },
    { taskID: 2, newTask: 'Task 2', geolocation: '', email: '', Image: '' },
    { taskID: 3, newTask: 'Task 3', geolocation: '', email: '', Image: '' },
    { taskID: 4, newTask: 'Task 4', geolocation: '', email: '', Image: '' },
    { taskID: 5, newTask: 'Task 5', geolocation: '', email: '', Image: '' }
];

var TodosService = /** @class */ (function () {
    /**
     * @param {?} _http
     */
    function TodosService(_http) {
        this._http = _http;
        this.URL = 'https://swapi.co/api/';
    }
    /**
     * @param {?} task
     * @return {?}
     */
    TodosService.prototype.createTask = function (task) {
        return TODOTASKS.push(task);
    };
    /**
     * @return {?}
     */
    TodosService.prototype.readTasks = function () {
        return TODOTASKS;
        /*
          return this._http.get(this.swapiURL + 'planets/')
            .map(res => {
                return new TodoTask(res);
              });
        */
    };
    /**
     * @param {?} task
     * @return {?}
     */
    TodosService.prototype.updateTask = function (task) {
        return TODOTASKS;
    };
    /**
     * @param {?} task
     * @return {?}
     */
    TodosService.prototype.deleteTask = function (task) {
        return TODOTASKS;
    };
    TodosService.decorators = [
        { type: Injectable },
    ];
    /**
     * @nocollapse
     */
    TodosService.ctorParameters = function () { return [
        { type: HttpClient, },
    ]; };
    return TodosService;
}());

var TodoTaskModule = /** @class */ (function () {
    function TodoTaskModule() {
    }
    /**
     * @return {?}
     */
    TodoTaskModule.forRoot = function () {
        return {
            ngModule: TodoTaskModule,
            providers: []
        };
    };
    TodoTaskModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule
                    ],
                    declarations: [
                        TodoTask,
                        TodosService
                    ],
                    exports: [
                        TodoTask,
                        TodosService
                    ]
                },] },
    ];
    /**
     * @nocollapse
     */
    TodoTaskModule.ctorParameters = function () { return []; };
    return TodoTaskModule;
}());

export { TodoTaskModule, TodoTask, TODOTASKS, TodosService };
