// todo-package/todo-package.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common'), require('@angular/common/http'), require('rxjs/add/operator/map')) :
	typeof define === 'function' && define.amd ? define(['exports', '@angular/core', '@angular/common', '@angular/common/http', 'rxjs/add/operator/map'], factory) :
	(factory((global['todo-package'] = {}),global.core,global.common,global.http));
}(this, (function (exports,core,common,http) { 'use strict';

var TodoTask = /** @class */ (function () {
    /**
     * @param {?=} init
     */
    function TodoTask(init) {
        Object.assign(this, init);
    }
    return TodoTask;
}());
var TODOTASKS = [
    { taskID: 1, newTask: 'Task 1', geolocation: '', email: '', Image: '' },
    { taskID: 2, newTask: 'Task 2', geolocation: '', email: '', Image: '' },
    { taskID: 3, newTask: 'Task 3', geolocation: '', email: '', Image: '' },
    { taskID: 4, newTask: 'Task 4', geolocation: '', email: '', Image: '' },
    { taskID: 5, newTask: 'Task 5', geolocation: '', email: '', Image: '' }
];

var TodosService = /** @class */ (function () {
    /**
     * @param {?} _http
     */
    function TodosService(_http) {
        this._http = _http;
        this.URL = 'https://swapi.co/api/';
    }
    /**
     * @param {?} task
     * @return {?}
     */
    TodosService.prototype.createTask = function (task) {
        return TODOTASKS.push(task);
    };
    /**
     * @return {?}
     */
    TodosService.prototype.readTasks = function () {
        return TODOTASKS;
        /*
          return this._http.get(this.swapiURL + 'planets/')
            .map(res => {
                return new TodoTask(res);
              });
        */
    };
    /**
     * @param {?} task
     * @return {?}
     */
    TodosService.prototype.updateTask = function (task) {
        return TODOTASKS;
    };
    /**
     * @param {?} task
     * @return {?}
     */
    TodosService.prototype.deleteTask = function (task) {
        return TODOTASKS;
    };
    TodosService.decorators = [
        { type: core.Injectable },
    ];
    /**
     * @nocollapse
     */
    TodosService.ctorParameters = function () { return [
        { type: http.HttpClient, },
    ]; };
    return TodosService;
}());

var TodoTaskModule = /** @class */ (function () {
    function TodoTaskModule() {
    }
    /**
     * @return {?}
     */
    TodoTaskModule.forRoot = function () {
        return {
            ngModule: TodoTaskModule,
            providers: []
        };
    };
    TodoTaskModule.decorators = [
        { type: core.NgModule, args: [{
                    imports: [
                        common.CommonModule
                    ],
                    declarations: [
                        TodoTask,
                        TodosService
                    ],
                    exports: [
                        TodoTask,
                        TodosService
                    ]
                },] },
    ];
    /**
     * @nocollapse
     */
    TodoTaskModule.ctorParameters = function () { return []; };
    return TodoTaskModule;
}());

exports.TodoTaskModule = TodoTaskModule;
exports.TodoTask = TodoTask;
exports.TODOTASKS = TODOTASKS;
exports.TodosService = TodosService;

Object.defineProperty(exports, '__esModule', { value: true });

})));
