// todo-package/todo-package.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common')) :
	typeof define === 'function' && define.amd ? define(['exports', '@angular/core', '@angular/common'], factory) :
	(factory((global['todo-package'] = {}),global.core,global.common));
}(this, (function (exports,core,common) { 'use strict';

var TodoTask = /** @class */ (function () {
    /**
     * @param {?=} init
     */
    function TodoTask(init) {
        Object.assign(this, init);
    }
    return TodoTask;
}());
var TODOTASKS = [
    { taskID: 1, newTask: 'Task 1', geolocation: '', email: '', Image: '' },
    { taskID: 2, newTask: 'Task 2', geolocation: '', email: '', Image: '' },
    { taskID: 3, newTask: 'Task 3', geolocation: '', email: '', Image: '' },
    { taskID: 4, newTask: 'Task 4', geolocation: '', email: '', Image: '' },
    { taskID: 5, newTask: 'Task 5', geolocation: '', email: '', Image: '' }
];

var TodoTaskModule = /** @class */ (function () {
    function TodoTaskModule() {
    }
    /**
     * @return {?}
     */
    TodoTaskModule.forRoot = function () {
        return {
            ngModule: TodoTaskModule,
            providers: []
        };
    };
    TodoTaskModule.decorators = [
        { type: core.NgModule, args: [{
                    imports: [
                        common.CommonModule
                    ],
                    declarations: [
                        TodoTask
                    ],
                    exports: [
                        TodoTask
                    ]
                },] },
    ];
    /**
     * @nocollapse
     */
    TodoTaskModule.ctorParameters = function () { return []; };
    return TodoTaskModule;
}());

exports.TodoTaskModule = TodoTaskModule;
exports.TodoTask = TodoTask;
exports.TODOTASKS = TODOTASKS;

Object.defineProperty(exports, '__esModule', { value: true });

})));
